doc/eval.md

```diff
@@ -110,3 +113,38 @@ The code above can be ran from a file:
       - age: 24
         city: 'Moscow'
 ```
+
+## Eval with retries
+
+If `cartridge_eval_with_retries` is set, that `eval` expects that function
+return `res, err`.
+It retries until `err` is received.
+
+It can be useful to perform some checks in Lua code and wait until some condition is met.
+
+Retries and delay can be configured:
+- `cartridge_eval_retries` (`number`, default: `3`) number of eval retries;
+- `cartridge_eval_delay` (`number`, default: `5`)  - eval retries delay;
+
+```yaml
+- name: Eval with retries
+  hosts: cluster
+  roles:
+    - ansible-cartridge
+  become: true
+  become_user: root
+  gather_facts: false
+  vars:
+    cartridge_scenario:
+      - eval
+    cartridge_eval_with_retries: true
+    cartridge_eval_retries: 10
+    cartridge_eval_delay: 1
+    cartridge_eval_body: |
+      local checkpoint_is_in_progress = box.info().gc().checkpoint_is_in_progress
+      if checkpoint_is_in_progress then
+        return nil, "Checkpoint is still in progress"
+      end
+
+      return true
```

Comment:
Maybe add some info in result? To show that eval_res is defined when retries.

```suggestion
      return ['connector', 'core']
```

Reply:
`return ['connector', 'core']` would fail. But I will think about it, `res, err` can be shown here.

Reply:
I can't think of any cases with using `eval_res` =(